internal/testing/types/bool.go

```diff
@@ -0,0 +1,57 @@
+package types
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-framework/attr"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-plugin-go/tftypes"
+)
+
+var (
+	_ attr.Type = BoolType{}
+)
+
+// BoolType is a reimplementation of types.BoolType that can be used as a base
+// for other extension types in testing.
+type BoolType struct{}
```

Comment:
Is there a reason the test types can't just be based off types.BoolType?

Reply:
If I recall, it was because I was having trouble extending a `primitive` value. I'll try again today.

Reply:
Here's where I was getting stuck.

Trying with type aliasing fails:

```go
type BoolTypeWithValidateError types.BoolType
```

```
types.BoolType (constant 2 of type types.primitive) is not a type (NotAType)
```

Creating a struct type (with pointer or concrete) fails:

```go
type BoolTypeWithValidateError struct {
	types.BoolType
}
```

```
types.BoolType (constant 2 of type types.primitive) is not a type (NotAType)
```

Trying as a variable also sensibly fails:

```go
var BoolTypeWithValidateError = types.BoolType

func (t BoolTypeWithValidateError) Validate(ctx context.Context, in tftypes.Value) []*tfprotov6.Diagnostic {
	return []*tfprotov6.Diagnostic{TestErrorDiagnostic}
}
```

```
BoolTypeWithValidateError (variable of type types.primitive) is not a type (NotAType)
```

Reply:
Annoying. :(

Reply:
I'll leave this information in a new issue for further consideration.